app/models/advocacy_campaign.rb

```diff
@@ -36,7 +34,8 @@ def validate_uniqueness
     end
 
     def set_identifiers
-      self.identifiers = [identifier] if identifier.present?
+      self.identifiers << "cta-aggregator:#{id}"
+      self.save(validate: false)
```

Comment:
1) It was just validated
2) The validation will fail because there are changes, but not on any of the fields that check uniqueness.